hack/check-everything.sh

```diff
@@ -74,7 +74,7 @@ function fetch_go_tools {
   header_text "Checking for gometalinter.v2"
   if ! is_installed golangci-lint; then
     header_text "Installing golangci-lint"
-    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.21.0
+    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.23.1
```

Comment:
long term maybe we introduce a tools.go to manage these dependencies with go modules - a few other tools do this.